src/Controller/AbsenceController.php

```diff
@@ -277,6 +490,37 @@ public function edit(Request $request)
         $this->templateParams(array('documents' => $this->getDocuments($a->id)));
         return $this->output('absences/edit.html.twig');
     }
+    /**
+     *@Route("/absence", name="absence.save", methods = {"POST"})
+     */
+    public function save_absence(Request $request) {
+
+        // Save absence(s).
+        $this->dbprefix = $GLOBALS['dbprefix'];
+        $this->droits = $GLOBALS['droits'];
+        $this->setAdminPermissions();
+
+        $result = $this->save($request, $this->admin);
+        $file = $request->files->get('documentFile');
+        if (!empty($file)) {
+            $token = $request->get("token");
+            if (!$this->isCsrfTokenValid('upload', $token)) {
+                return new Response("Operation not allowed",  Response::HTTP_BAD_REQUEST,['content-type' => 'text/plain']);
+            }
+            $filename = $file->getClientOriginalName();
+            $ad = new AbsenceDocument();
+            $ad->absence_id($result['id']);
+            $ad->filename($filename);
+            $ad->date(new \DateTime());
+            $this->entityManager->persist($ad);
+            $this->entityManager->flush();
+            $file->move(__DIR__ . AbsenceDocument::UPLOAD_DIR . $result['id'] . '/' . $ad->id(), $filename);
+        }
+
+        return $this->redirectToRoute('absence.index');
```

Comment:
Here we lost the success or error message. Please set the flashBag according to the result of saving